src/nvim/os/env.c

```diff
@@ -96,32 +83,65 @@ const char *os_getenv(const char *name)
     // except when it does not include the NUL-terminator.
     e = xmemdupz(buf, size);
   }
-  pmap_put(cstr_t)(&envmap, xstrdup(name), e);
 end:
   if (r != 0 && r != UV_ENOENT && r != UV_UNKNOWN) {
     ELOG("uv_os_getenv(%s) failed: %d %s", name, r, uv_err_name(r));
   }
   return e;
 }
 
+/// Like getenv(), but returns a pointer to `NameBuff` instead of allocating.
+/// Value is truncated if it exceeds sizeof(NameBuff).
+/// @see os_env_exists
+char *os_getenv_noalloc(const char *name)
+  FUNC_ATTR_NONNULL_ALL
+{
+  if (name[0] == NUL) {
+    return NULL;
+  }
+
+  size_t size = sizeof(NameBuff);
+  int r = uv_os_getenv(name, NameBuff, &size);
+  if (r == UV_ENOBUFS) {
+    char *e = xmalloc(size);
+    r = uv_os_getenv(name, e, &size);
+    if (r == 0 && size != 0 && e[0] != NUL) {
+      xmemcpyz(NameBuff, e, sizeof(NameBuff) - 1);
+    }
+    xfree(e);
+  }
+
+  if (r != 0 || size == 0 || NameBuff[0] == NUL) {
+    if (r != 0 && r != UV_ENOENT && r != UV_UNKNOWN) {
+      ELOG("uv_os_getenv(%s) failed: %d %s", name, r, uv_err_name(r));
+    }
+    return NULL;
+  }
+  return NameBuff;
+}
+
 /// Returns true if environment variable `name` is defined (even if empty).
 /// Returns false if not found (UV_ENOENT) or other failure.
-bool os_env_exists(const char *name)
+///
+/// @param name the environment variable in question
+/// @param nonempty whether the variable has a valid value, or may be empty
+/// @return whether the variable exists (and is defined)
```

Comment:
this is  very confusing. "defined" and "exists" are synonyms usually. 

```suggestion
/// @return whether the variable exists (and is nonempty)
```

Reply:
I agree, I was much struggling with the terminology here. Suggestion added as proposed.